modules/ocf/functions/lookup_wrap.pp

```diff
@@ -0,0 +1,8 @@
+function ocf::lookup_wrap($item) {
+  if String($item) =~ /^[a-zA-Z_:]*$/ {
```

Comment:
This is a little hacky. I'd like to fail harder here since we are dealing with passwords. It's a little tricky because all the values that get passed in are hashes, even the keys.

Reply:
I would match on type instead:
```puppet
function ocf::lookup_wrap($item) {
  $item ? {
    String  => lookup($item),
    default => $item,
  }
}
```